src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +66,15 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation)
+                    {
+                        Pop(history[history.Count-2]);
```

Comment:
Slightly unrelated, but I'd prefer not to use `Debug.Assert` and instead throw an exception. `Debug.Assert` has been making my life a misery lately: https://github.com/github/VisualStudio/issues/774